internal/db/db.go

```diff
@@ -0,0 +1,48 @@
+package db
+
+import (
+	"context"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/stellar/go/support/errors"
+)
+
+type DBConnectionPool interface {
+	Close() error
+	Ping() error
+	DriverName() string
+	sqlx.ExecerContext
+	sqlx.QueryerContext
+	sqlx.PreparerContext
+	GetContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+	SelectContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+}
```

Comment:
❓ Is it not clear to me why we need to create an interface here? Can you explain a little bit more? Is it for testing purposes? Do we plan on swapping to another implementation in the future?

Reply:
The idea with the interface is that it would make it very trivial to add query metrics in the future.
As discussed on standup, we will look into what instruments AWS RDS provides in that respect to make a decision about keeping the interface vs. using any existing RDS tools.

Reply:
https://cheesecake.slack.com/archives/CJ55HPNR4/p1714421061072399

Reply:
Keep in mind that this is a public repo. People wouldn't be able to see what that link is about.

What would be the thing that you'd like to track on top of query execution time?

Another thing to consider is whether it will be easy for people to configure their own query metric-tracking system.

Reply:
I've removed the Slack thread link. No links should live in the open repo indeed, not so much for lack of access but for security reasons I would say.

I don't have a particular array of metrics I would like to implement off the bat, but the idea would be to keep the implementation flexible for whenever ideas and requirements for this arise, making it easier to integrate them without major refactors.

But query time (planning and execution) seems like a good first candidate. One thing the folks from the OPS team mentioned in the Slack thread was the possibility of hooking up alerts here as well, which seems like a great option to have.